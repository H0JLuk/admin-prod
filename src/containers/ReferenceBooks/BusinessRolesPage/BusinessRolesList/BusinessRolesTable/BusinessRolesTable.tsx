import React from 'react';
import { Table, TableProps } from 'antd';
import EmptyMessage from '@components/EmptyMessage';
import { getFormattedDate } from '@utils/helper';
import { BusinessRoleDto } from '@types';

import styles from './BusinessRolesTable.module.css';

type IBusinessRolesListTable = {
    businessRolesList: BusinessRoleDto[];
    rowSelection: TableProps<BusinessRoleDto>['rowSelection'];
    onRow: TableProps<BusinessRoleDto>['onRow'];
    loading: boolean;
};

const { Column } = Table;

const BUSINESS_ROLES_NAME = 'Название';
const BUSINESS_ROLES_ID = 'Начало действия';

const BusinessRolesTable: React.FC<IBusinessRolesListTable> = ({
    businessRolesList,
    rowSelection,
    onRow,
    loading,
}) => (
    <div className={styles.tableWrapper}>
        <Table
            loading={loading}
            showHeader={false}
            dataSource={businessRolesList}
            rowKey="id"
            className={styles.table}
            rowSelection={rowSelection}
            onRow={onRow}
            locale={{ emptyText: <EmptyMessage /> }}
            pagination={false}
        >
            <Column width="80%" dataIndex="name" render={(item) => columnRender(item, BUSINESS_ROLES_NAME)} />
            <Column width="20%" dataIndex="startDate" render={(item) => columnRender(getFormattedDate(item), BUSINESS_ROLES_ID)} />
        </Table>
    </div>
);

const columnRender = (item: string, name: string) => (
    <>
        <div className={styles.tableTitle}>{name}</div>
        <div className={styles.tableCell}>{item}</div>
    </>
);

export default BusinessRolesTable;
