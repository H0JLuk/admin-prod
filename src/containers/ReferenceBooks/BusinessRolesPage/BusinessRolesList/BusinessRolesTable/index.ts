import BusinessRolesTable from './BusinessRolesTable';

export default BusinessRolesTable;
