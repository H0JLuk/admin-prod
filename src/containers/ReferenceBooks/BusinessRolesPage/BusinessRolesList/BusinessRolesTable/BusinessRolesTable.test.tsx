import React from 'react';
import { shallow } from 'enzyme';
import BusinessRolesTable from './BusinessRolesTable';

const TEST_ROLES_LIST = [
    {
        id: 1,
        name: 'TEST',
        description: 'TEST_DESCRIPTION',
        deleted: false,
        startDate: '2019-12-01',
        endDate: '2021-09-11',
    },
];

const BusinessRolesTableTestProps = {
    businessRolesList: TEST_ROLES_LIST,
    rowSelection: {},
    onRow: () => ({}),
    loading: false,
};

describe('BusinessRoleTable test', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<BusinessRolesTable {...BusinessRolesTableTestProps} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
