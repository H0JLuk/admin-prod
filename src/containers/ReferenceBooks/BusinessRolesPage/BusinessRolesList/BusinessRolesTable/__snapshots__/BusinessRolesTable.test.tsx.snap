// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessRoleTable test should match snapshot 1`] = `
"<div className=\\"tableWrapper\\">
  <Table loading={false} showHeader={false} dataSource={{...}} rowKey=\\"id\\" className=\\"table\\" rowSelection={{...}} onRow={[Function: onRow]} locale={{...}} pagination={false}>
    <Column width=\\"80%\\" dataIndex=\\"name\\" render={[Function: render]} />
    <Column width=\\"20%\\" dataIndex=\\"startDate\\" render={[Function: render]} />
  </Table>
</div>"
`;
